connectors/amazons3/connector/src/main/java/org/apache/manifoldcf/crawler/connectors/amazons3/AmazonS3Connector.java

```diff
@@ -0,0 +1,917 @@
+
+  private static final String BUCKET_SPLITTER = ",";
+
+  private static final String TAB_NAME = "TabName";
+
+  private static final String SELECTED_NUM = "SelectedNum";
+
+  private static final String SEQ_NUM = "SeqNum";
+
+  protected final static String ACTIVITY_READ = "read document";
+
+  protected long lastSessionFetch = -1L;
+
+  protected static final long timeToRelease = 300000L;
+
+  protected AmazonS3 amazonS3;
+
+  protected boolean connected = false;
+
+  protected String amazons3ProxyHost = null;
+
+  protected String amazons3ProxyPort = null;
+
+  protected String amazons3ProxyDomain = null;
+
+  protected String amazons3ProxyUserName = null;
+
+  protected String amazons3ProxyPassword = null;
+
+  protected String amazons3AwsAccessKey = null;
+
+  protected String amazons3AwsSecretKey = null;
+
+  private static final String STD_SEPARATOR_BUCKET_AND_KEY = BUCKET_SPLITTER;
+
+  private String[] buckets;
+
+  private DocumentProcess documentProcess;
+
+  public AmazonS3Connector() {
+    super();
+    documentProcess = new GenericDocumentProcess();
+  }
+
+  @Override
+  public String[] getActivitiesList() {
+    return new String[] { ACTIVITY_READ };
+  }
+
+  @Override
+  public String[] getBinNames(String documentIdentifier) {
+    return new String[] { amazons3AwsAccessKey };
+  }
+
+  /**
+   * Close the connection. Call this before discarding the connection.
+   */
+  @Override
+  public void disconnect() throws ManifoldCFException {
+    amazons3AwsAccessKey = null;
+    amazons3AwsSecretKey = null;
+
+    amazons3ProxyHost = null;
+    amazons3ProxyPort = null;
+    amazons3ProxyDomain = null;
+    amazons3ProxyUserName = null;
+    amazons3ProxyPassword = null;
+  }
+
+  /**
+   * Connect method initializes the configparams
+   * */
+  @Override
+  public void connect(ConfigParams configParams) {
+    super.connect(configParams);
+
+    // aws access and secret keys
+    amazons3AwsAccessKey = configParams
+        .getParameter(AmazonS3Config.AWS_ACCESS_KEY);
+    amazons3AwsSecretKey = configParams
+        .getObfuscatedParameter(AmazonS3Config.AWS_SECRET_KEY);
+
+    // proxy values
+    amazons3ProxyHost = configParams
+        .getParameter(AmazonS3Config.AMAZONS3_PROXY_HOST);
+    amazons3ProxyPort = configParams
+        .getParameter(AmazonS3Config.AMAZONS3_PROXY_PORT);
+    amazons3ProxyDomain = configParams
+        .getParameter(AmazonS3Config.AMAZONS3_PROXY_DOMAIN);
+    amazons3ProxyUserName = configParams
+        .getParameter(AmazonS3Config.AMAZONS3_PROXY_USERNAME);
+    amazons3ProxyPassword = configParams
+        .getObfuscatedParameter(AmazonS3Config.AMAZONS3_PROXY_PASSWORD);
+  }
+
+  /**
+   * Get the Amazons3 client, relevant access keys should have been posted
+   * already
+   * @return
+ * @throws ManifoldCFException 
+   */
+  protected AmazonS3 getClient() throws ManifoldCFException {
+    if (amazonS3 == null) {
+      try {
+        BasicAWSCredentials awsCreds = new BasicAWSCredentials(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

If possible, use temporary security credentials (IAM roles) instead of long-term access keys.
Long-term access keys, such as those associated with IAM users and AWS account root users, remain valid until you manually revoke them. However, temporary security credentials obtained through IAM roles and other features of the AWS Security Token Service expire after a short period of time. Use temporary security credentials to help reduce your risk in case credentials are accidentally exposed.
Learn more about [best practices for managing AWS access keys](https://docs.aws.amazon.com/general/latest/gr/aws-access-keys-best-practices.html).